### PR TITLE
MueLu: Update License and Copyright

### PR DESCRIPTION
**Pull request by ccober6**
Update the License and Copyright files to
 * Reflect NTESS and new contract
 * Utilize SPDX identifiers within files
 * Make copyright and license files consistent across Trilinos packages (i.e., LICENSE and COPYRIGHT)

@trilinos/muelu

### CODE REVIEW

**Review by jhux2 — CHANGES_REQUESTED** [review]
@ccober6 Could you remove individual names from the copyright notice and keep just the MueLu team github handle?

**Review by mayrmt — COMMENTED** [review]
The year in the copyright statement is 2012. Is this intentional back-dated? While some code might be from back then, lots of code is more recent.

File: packages/muelu/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+
+## Questions? 
+Contact lead developers:
+
+* MueLu team         (GitHub handle: @trilinos/muelu)
+* Jonathan Hu        (GitHub handle: [jhux2](https://github.com/jhux2) or jhu@sandia.gov)
+* Andrey Prokopenko  (GitHub handle: [aprokop](https://github.com/aprokop)
+* Ray Tuminaro       (GitHub handle: [rstumin](https://github.com/rstumin) or rstumin@sandia.gov)
+* Tobias Wiesner     (GitHub handle: [tawiesn](https://github.com/tawiesn)
```
Comment (mayrmt) [inline packages/muelu/README.md:15]: Andrey and Tobias are not workin on MueLu anymore.

@jhux2 Is this the place, where you'd prefer to just put the MueLu team GitHub handle?

Comment (jhux2) [reply packages/muelu/README.md:15]: @mayrmt That's right.  I see that @ccober6 already removed individual names from the other headers, which is great.  (I had been meaning to do that but never got around to it.)

Comment (ccober6) [reply packages/muelu/README.md:15]: @mayrmt and @jhux2 , yeah, I can just leave the muelu handle.  There was not another copyright assertion done, so just leaving the previous copyright in-place.  This is just cleaning up the copyright notice and license, along with including the SPDX-License-Identifier.

Comment (ccober6) [reply packages/muelu/README.md:15]: FYI, by cleaning up the headers, this was able to remove +30k lines of comments!

Comment (jhux2) [reply packages/muelu/README.md:15]: @ccober6 You are going to jump into the Trilinos contributor lead for sure!

Comment (ccober6) [reply packages/muelu/README.md:15]: Yeah, I will have touched almost every file and eliminated a lot of comment lines.  The Great Comment Killer!  :)

**Review by jhux2 — APPROVED** [review]
@ccober6 Looks good, thank you.

**Review by mayrmt — APPROVED** [review]

**Comment by cgcgcg** [comment]
@ccober6 This PR broke the clang-format check.

**Pull request merged**